container/container.go

```diff
@@ -42,6 +43,9 @@ type SubcontainerEvent struct {
 
 	// The full container name of the container where the event occurred.
 	Name string
+
+	//who detected it
+	DetectionType string
```

Comment:
I still don't like having "detection type". It just feels messy and not the right abstraction. Why is it necessary? I don't think container factories should care how the container was detected, their job is to just create the handler for it. It should be up to the manager (or a child module) to ensure that the correct factories are hooked up to the right detection sources.